drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -185,7 +185,7 @@ class SingleTimeKinematicConstraintWrapper : public Constraint
     rigid_body_constraint->getRobotPointer()->doKinematics(kinsol);
     VectorXd y; MatrixXd dy;
     rigid_body_constraint->eval(nullptr,kinsol,y,dy);
-    initTaylorVecXd(y,dy*autoDiffToGradientMatrix(tq),ty);
+    initializeAutoDiffGivenGradientMatrix(y, (dy * autoDiffToGradientMatrix(tq)).eval(), ty);
```

Comment:
by the way, i don't really like that i have to do this everywhere (bring the chain rule inside of my method).  i thought a little about method signatures which took a double in and outputted an autodiff (so the chain rule would be on the outside), but ultimately decided this was cleaner; I think.  When it's templated, everything will work correctly (except that people have to write their templated methods using that ugly cast to scalartype for data matrices).  But I worry that wrapper methods like this will incur bugs where people forget the chain rule component.